sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -302,19 +302,27 @@ public long putLog(@NonNull Log log, @NonNull String group, @IntRange(from = Fla
             if (maxSize == OPERATION_FAILED_FLAG) {
                 throw new PersistenceException("Failed to store a log to the Persistence database.");
             }
-            if (!isLargePayload && maxSize <= payloadSize) {
+            if (maxSize <= payloadSize) {
                 throw new PersistenceException("Log is too large (" + payloadSize + " bytes) to store in database. " +
                         "Current maximum database size is " + maxSize + " bytes.");
             }
             int priority = Flags.getPersistenceFlag(flags, false);
             contentValues = getContentValues(group, isLargePayload ? null : payload, targetToken, log.getType(), targetKey, priority);
+            long storedDataSize = getStoredDataSize();
+            while (payloadSize + getStoredDataSize() > maxSize)
+            {
+                AppCenterLog.debug(LOG_TAG, "Storage is full, trying to delete the oldest log that has the lowest priority which is lower or equal priority than the new log.");
```

Comment:
Potentially this log can be duplicated many times in the output. Let's add a single log before while. Probably wrap it into a `if (isLargePayload)` condition.

Reply:
Discussed with @MatkovIvan and decided that it is OK. Resolving.